integration-tests/hardhat.config.ts

```diff
@@ -18,7 +18,7 @@ const config: HardhatUserConfig = {
     timeout: isLiveNetwork() ? 300_000 : 75_000,
   },
   solidity: {
-    version: '0.8.9',
+    version: '0.7.6',
```

Comment:
I would recommend using hardhat's native system for applying multiple compiler types so we can use 0.8.9 (or whatever the latest is at any given time) for most contracts and only apply older versions to contracts that really must have it. You can find a guide here: https://hardhat.org/guides/compile-contracts.html#multiple-solidity-versions

Reply:
^^ Will also cut down the diff in this PR.

Reply:
Will do. Was unaware of that feature, thanks for letting me know.